### PR TITLE
Refactor available modals with Global Context

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x] GlobalContext
- [x] TransferProcessModal [Deprecated]
- [x] PositionDetailsModal
- [x] OpenSubTab
- [x] OpenPositionItem
- [x] WalletPanel
- [x] 128   LoadingModal
- [x] 129 	SuccessfulModal
- [x] 130	CanceledModal
- [x] 131	FailedModal
- [x] 005	DepositModal
- [x] 006	WithdrawalModal

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 1
react-usestateref: v1.0.8
- new Class / Component: 6
src/components/canceled_modal/canceled_modal.tsx
src/components/deposit_modal/deposit_modal.tsx
src/components/failed_modal/failed_modal.tsx
src/components/loading_modal/loading_modal.tsx
src/components/successful_modal/successful_modal.tsx
src/components/withdrawal_modal/withdrawal_modal.tsx
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/contexts/global_context.tsx [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+
+export interface IGlobalContext {
+  width: number;
+  height: number;
+  layoutAssertion: LayoutAssertionUnion;
+  initialColorMode: ColorModeUnion;
+  colorMode: ColorModeUnion;
+  toggleColorMode: () => void;
+  toast: (props: IToastify) => void;
+  visiblePositionDetailsModal: boolean;
```
Comment (arealclimber) [inline src/contexts/global_context.tsx:87]: 空格

File: src/contexts/global_context.tsx [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+const initialColorMode: ColorModeUnion = 'dark';
+
+export const GlobalProvider = ({children}: IGlobalProvider) => {
+  const [colorMode, setColorMode] = useState<ColorModeUnion>(initialColorMode);
+
+  const [visiblePositionDetailsModal, setVisiblePositionDetailsModal] = useState(false);
+  const [dataPositionDetailsModal, setDataPositionDetailsModal] =
+    useState<IDataPositionDetailsModal>({openCfdDetails: dummyOpenCFDDetails});
+
+  // TODO: transfer
```
Comment (arealclimber) [inline src/contexts/global_context.tsx:178]: 刪註解

File: src/contexts/global_context.tsx [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+});
+
+const initialColorMode: ColorModeUnion = 'dark';
+
+export const GlobalProvider = ({children}: IGlobalProvider) => {
+  const [colorMode, setColorMode] = useState<ColorModeUnion>(initialColorMode);
+
+  const [visiblePositionDetailsModal, setVisiblePositionDetailsModal] = useState(false);
+  const [dataPositionDetailsModal, setDataPositionDetailsModal] =
+    useState<IDataPositionDetailsModal>({openCfdDetails: dummyOpenCFDDetails});
```
Comment (arealclimber) [inline src/contexts/global_context.tsx:176]: 可以塞 initialData

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**